packages/core/http/core-http-router-server-internal/src/versioned_router/route_version_utils.ts

```diff
@@ -38,3 +46,34 @@ export function isValidRouteVersion(isPublicApi: boolean, version: string): unde
     ? undefined
     : `Invalid version number. Received "${version}", expected a string containing _only_ a finite, whole number greater than 0.`;
 }
+
+function hasQueryVersion(request: KibanaRequest) {
+  return isObject(request.query) && ELASTIC_HTTP_VERSION_QUERY_PARAM in request.query;
+}
+export function hasVersion(request: KibanaRequest, isQueryVersionEnabled?: boolean): boolean {
+  return Boolean(
+    ELASTIC_HTTP_VERSION_HEADER in request.headers ||
+      (isQueryVersionEnabled && hasQueryVersion(request))
+  );
+}
+
+export function redactQueryVersion(request: Mutable<KibanaRequest>): void {
+  if (hasQueryVersion(request)) {
+    request.query = omit({ ...(request.query as {}) }, ELASTIC_HTTP_VERSION_QUERY_PARAM);
```

Comment:
NIT: can't we just `delete request.query[ELASTIC_HTTP_VERSION_QUERY_PARAM]` (or is the object hard read-only?) It would avoid a a double instantiation (`...` + `omit`)